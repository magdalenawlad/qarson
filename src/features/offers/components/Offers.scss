@use '../../../styles/variables' as v;

.offers-container {
  width: 100%;

  .offers-label {

    span {
      font-family: 'LatoBlack';
    }
  }

  .offers-list {
    padding: 2rem 0;;
    display: grid;
    grid-template-columns: repeat(auto-fill, 300px);
    grid-gap: 2rem;
    justify-content: center;
  }
}

.offer-card {
  background: transparent;
  border-radius: 1rem;;
  box-shadow: 3px 3px 10px 2px rgba(v.$textPrimary, 0.3);
  display: flex;
  align-items: center;
  flex-direction: column;
  overflow: hidden;
  position: relative;
  @include v.shadow();

  img {
    width: auto;
    width: 100%;
    height: 168px;
  }

  &__description {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    width: 100%;
    padding: 1rem;

    &__name {
      font-family: 'LatoBlack';
    }
  }

  &__footer {
    display: flex;
    width: 100%;
  
    > div, > button {
      flex: 1;
    }

    .offer-card__footer__button {
      display: flex;
      justify-content: center;
      align-items: center;
      background: v.$textPrimary;
      color: v.$background;
      height: 3rem;
      border: none;
      border-radius: 1rem 0 0;
      font-size: v.$fontSizeRegular;
      cursor: pointer;

      &--selected {
        background: v.$qarsonPrimary;
        color: v.$textPrimary;
        border-radius: 1rem 0 0;
        box-shadow: inset 7px 7px 14px #6e872a, inset -7px -7px 14px #d6ff52;
      }

      &:disabled {
        opacity: 0.4;
        cursor: default;
      }

      svg {
        margin-right: 0.3rem;
      }
    }

    &__price {
      display: flex;
      align-items: center;
      justify-content: center;
      font-family: 'LatoBlack';
    }
  }
  
  .indicator {
    margin: 0;
    opacity: 0.4;
    padding: 0;
    background: v.$textPrimary;
    color: v.$background;
    padding: 0.5rem 0;
    position: absolute;
    top:0;
    right:0;
    transform: translateX(30%) translateY(0%) rotate(45deg);
    transform-origin: top left;
  }

  .indicator:before,
  .indicator:after {
    content: '';
    position: absolute;
    top:0;
    margin: 0 -1px;
    width: 100%;
    height: 100%;
    background: v.$textPrimary;
  }

  .indicator:before {
    right:100%;
  }
  
  .indicator:after {
    left:100%;
  }

  .car-placeholder {
    padding: 3rem;
    opacity: 0.5
  }
}