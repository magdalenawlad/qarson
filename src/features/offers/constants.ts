import vwLogo from '../../assets/images/carLogos/vw.svg';
import seatLogo from '../../assets/images/carLogos/seat.svg';
import renaultLogo from '../../assets/images/carLogos/renault.png';
import peugeotLogo from '../../assets/images/carLogos/peugeot.png';
import kiaLogo from '../../assets/images/carLogos/kia.png';
import skodaLogo from '../../assets/images/carLogos/skoda.png';

type EnumDictionary<T extends string | symbol | number, U> = {
  [K in T]: U;
};

export enum Make {
  Volkswagen = 'Volkswagen',
  Skoda = 'Skoda',
  Seat = 'Seat',
  Renualt ='Renualt',
  Pegeuot = 'Pegeuot',
  Kia = 'Kia'
}

export const MAKE_LOGO_PATH: EnumDictionary<Make, string> = {
  [Make.Volkswagen]: vwLogo,
  [Make.Skoda]: skodaLogo,
  [Make.Seat]: seatLogo,
  [Make.Renualt]: renaultLogo,
  [Make.Pegeuot]: peugeotLogo,
  [Make.Kia]: kiaLogo
};
