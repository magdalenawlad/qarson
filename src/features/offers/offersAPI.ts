import { response } from "./mockOffersResponse";
import { Offer } from "./offersSlice";

export interface FetchDataResponse {
  answer: string;
  dataGeneration: string;
  offers: Offer[];
}

// mock async request for offers data
export const fetchData = () => (
  new Promise<FetchDataResponse>((resolve) =>
    setTimeout(() => resolve(response), 1000)
  )
);
