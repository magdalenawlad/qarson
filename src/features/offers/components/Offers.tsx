import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '../../../app/store';
import { Offer } from './Offer';
import { OffersSearch } from './OffersSearch';
import './Offers.scss';
import { Make } from '../constants';

export const Offers = () => {
  const data = useSelector((state: RootState) => state.offers.data);
  const selectedOffers = useSelector((state: RootState) => state.offers.selected);
  const filters = useSelector((state: RootState) => state.offers.filters);
  const isLoading = useSelector((state: RootState) => state.offers.isLoading);

  const filteredData = () => {
    let result = data;
    const makeFiltersApplied = filters.make.find(({ selected }) => selected);

    if (makeFiltersApplied) {
      result = result.filter(({ make }) => {
        const searched = filters
          .make.find((({ name }) => {
            return name === make;
          }));

        return searched?.selected
      });
    }

    if (filters.text) {
      const { text } = filters;
      result = result.filter(({ make, model }) =>
        Make[make].toLowerCase().includes(text.toLowerCase())
          || model.toLowerCase().includes(text.toLowerCase()));
    }

    if (filters.showOnlyAvailable) {
      result = result.filter(({ availability }) => !!availability);
    }

    return result;
  };

  const carsToShow = filteredData();

  return (
    <>
      <OffersSearch />
      <div className='offers-container'>
        {isLoading ? (
          <div>Loading...</div>
        ): (
          <div className='offers'>
            {carsToShow && (
              <div className='offers-label'>
                Offers found: <span>{carsToShow.length}</span>
              </div>
            )}
            <div className='offers-list'>
              {carsToShow && carsToShow.map((item) => (
                <Offer
                  key={`item-${item.id}`}
                  selected={selectedOffers.indexOf(item.id) >=0}
                  {...item}
                />
              ))}
            </div>
          </div>
        )}
      </div>
    </>
  )
};
