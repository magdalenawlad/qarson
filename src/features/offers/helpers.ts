import { MAKE_LOGO_PATH, Make } from './constants';

export const getMakeLogoPath = (make: Make) => MAKE_LOGO_PATH[make];