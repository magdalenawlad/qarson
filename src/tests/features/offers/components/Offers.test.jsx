import React from 'react';
import { screen } from '@testing-library/react';
import { renderWithProviders, initialState } from '../../../test-utils';
import { Offers } from '../../../../features/offers/components/Offers.tsx';
import { response } from '../../../../features/offers/mockOffersResponse';
import { Make } from '../../../../features/offers/constants';

const preloadedStateDefault = {
  ...initialState,
  offers: {
    ...initialState.offers,
    data: response.offers
  }
};

test('renders correctly', () => {
  const preloadedState = preloadedStateDefault;

  const {container } = renderWithProviders(<Offers />, { preloadedState });

  expect(container).toMatchSnapshot();
});

test('displays loading indicator when in loading state', () => {
  const preloadedState = {
    ...preloadedStateDefault,
    offers: {
      ...preloadedStateDefault.offers,
      isLoading: true
    }
  };

  renderWithProviders(<Offers />, { preloadedState });

  expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
});

test('displays correct message when no offers found', () => {
  const preloadedState = {
    ...preloadedStateDefault,
    offers: {
      ...preloadedStateDefault.offers,
      data: []
    }
  };
  renderWithProviders(<Offers />, { preloadedState });

  expect(screen.getByText(/Offers found:/i)).toBeInTheDocument();
  expect(screen.getByText(/0/i)).toBeInTheDocument();
});

test('displays correct number of offers', () => {
  const preloadedState = preloadedStateDefault;

  renderWithProviders(<Offers />, { preloadedState });

  expect(screen.getByText(/Offers found:/i)).toBeInTheDocument();
  expect(screen.getByText(/11/i)).toBeInTheDocument();
  expect(screen
    .queryAllByTestId('offer-card'))
    .toHaveLength(11);
});

test('displays correct number of offers when search filter applied', () => {
  const preloadedState = {
    ...preloadedStateDefault,
    offers: {
      ...preloadedStateDefault.offers,
      filters: {
        ...preloadedStateDefault.offers.filters,
        text: 'vol'
      }
    }
  };

  renderWithProviders(<Offers />, { preloadedState });

  expect(screen
    .queryAllByTestId('offer-card'))
    .toHaveLength(3);
});

test('displays correct number of offers when make quick filter applied', () => {
  const preloadedState = {
    ...preloadedStateDefault,
    offers: {
      ...preloadedStateDefault.offers,
      filters: {
        ...preloadedStateDefault.offers.filters,
        make: [
          { name: Make.Seat, selected: true },
          { name: Make.Pegeuot, selected: true },
          { name: Make.Volkswagen, selected: true }
        ]
      }
    }
  };

  renderWithProviders(<Offers />, { preloadedState });

  expect(screen
    .queryAllByTestId('offer-card'))
    .toHaveLength(6);
});

test('displays correct number of offers when availability filter aplied', () => {
  const preloadedState = {
    ...preloadedStateDefault,
    offers: {
      ...preloadedStateDefault.offers,
      filters: {
        ...preloadedStateDefault.offers.filters,
        showOnlyAvailable: true
      }
    }
  };

  renderWithProviders(<Offers />, { preloadedState });

  expect(screen
    .queryAllByTestId('offer-card'))
    .toHaveLength(6);
});
