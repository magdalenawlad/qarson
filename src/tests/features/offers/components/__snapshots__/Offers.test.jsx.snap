// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="search"
  >
    <div>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root search-textfield css-wb57ya-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            id="quick-filter-text"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-hdw1oc"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="search-quick-filters"
    >
      <div
        class="availability-filter"
      >
        <span
          class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary availability-filter__checkbox css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
        >
          <input
            class="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate="false"
            type="checkbox"
            value="false"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <div
          class="availability-filter__label"
        >
          Available now
        </div>
      </div>
    </div>
  </div>
  <div
    class="offers-container"
  >
    <div
      class="offers"
    >
      <div
        class="offers-label"
      >
        Offers found: 
        <span>
          11
        </span>
      </div>
      <div
        class="offers-list"
      >
        <div
          class="offer-card offer-card--disabled"
          data-testid="offer-card"
        >
          <div
            class="indicator"
          >
            Unavailable
          </div>
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/60653/mandataire-auto-cd-fl030-ba75c5-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Volkswagen Golf VII
            </div>
            <div>
              1.6 TDI
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card "
          data-testid="offer-card"
        >
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/51897/mandataire-auto-cd-fl030-1c5b73-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Volkswagen up!
            </div>
            <div>
              1.0 BMT
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card offer-card--disabled"
          data-testid="offer-card"
        >
          <div
            class="indicator"
          >
            Unavailable
          </div>
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/51897/mandataire-auto-cd-fl030-1c5b73-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Volkswagen Polo VI
            </div>
            <div>
              1.0 TSI
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card "
          data-testid="offer-card"
        >
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/51250/face-arriere-droite-karoq-cd-lr120-e8b54a-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Skoda Karoq
            </div>
            <div>
              1.6 TDI
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card offer-card--disabled"
          data-testid="offer-card"
        >
          <div
            class="indicator"
          >
            Unavailable
          </div>
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/41392/face-avant-gauche-30-deg-arona-lf30-5b531f-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Seat Arona
            </div>
            <div>
              1.0 TSI
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card "
          data-testid="offer-card"
        >
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/62030/mandataire-auto-cd-fl030-cfca76-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Renualt Twingo
            </div>
            <div>
              1.0 Sce
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card offer-card--disabled"
          data-testid="offer-card"
        >
          <div
            class="indicator"
          >
            Unavailable
          </div>
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/62030/mandataire-auto-cd-fl030-cfca76-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Renualt Clio
            </div>
            <div>
              1.0 Tce
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card "
          data-testid="offer-card"
        >
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/54791/mandataire-auto-cd-fl030-af72bd-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Pegeuot 208
            </div>
            <div>
              1.2 PureTech
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card offer-card--disabled"
          data-testid="offer-card"
        >
          <div
            class="indicator"
          >
            Unavailable
          </div>
          <img
            alt="Auto"
            class="car-placeholder"
            data-testid="offer-photo"
            src="car-placeholder.png"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Pegeuot 208
            </div>
            <div>
              1.2 PureTech
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
              disabled=""
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card "
          data-testid="offer-card"
        >
          <img
            alt="Auto"
            class="car-placeholder"
            data-testid="offer-photo"
            src="car-placeholder.png"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Kia Rio
            </div>
            <div>
              1.25 CVVT
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
        <div
          class="offer-card "
          data-testid="offer-card"
        >
          <img
            alt="Auto"
            data-testid="offer-photo"
            src="https://cars.qarson.fr/64881/mandataire-auto-cd-fl030-29e909-bcp.jpg"
          />
          <div
            class="offer-card__description"
          >
            <div
              class="offer-card__description__name"
            >
              Kia Rio
            </div>
            <div>
              1.25 CVVT
            </div>
          </div>
          <div
            class="offer-card__footer"
          >
            <div
              class="offer-card__footer__price"
            >
              $559.99
            </div>
            <button
              class="offer-card__footer__button "
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 9h2V6h3V4h-3V1h-2v3H8v2h3v3zm-4 9c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-9.83-3.25.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.86-7.01L19.42 4h-.01l-1.1 2-2.76 5H8.53l-.13-.27L6.16 6l-.95-2-.94-2H1v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.13 0-.25-.11-.25-.25z"
                />
              </svg>
              Add
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
